File: pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import com.pinyougou.group.Cart;
import com.pinyougou.pojo.TbAddress;

import java.util.List;

public interface CartService {
    void saveCartListToRedis(String username, List<Cart> cartList);

    List<Cart> addToCartList(Long itemId, int num, List<Cart> cartList);

    List<Cart> getCartListFronRedis(String username);

    List<Cart> mergeCartList(List<Cart> cookieCartList, List<Cart> redisCartList);

    List<TbAddress> findAddressList(String username);
}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
app.service("cartService",function ($http) {
    //1.查询购物车列表
    this.findCartList = ()=>{
        return $http.get("./cart/getCartList.do");
    };
    //2.添加到购物车
    this.addGoodsToCartList=(itemId,num)=>{
        return $http.get("./cart/addToCartList.do?itemId="+itemId+"&num="+num);
    };
    //3.查询用户的收货地址列表
    this.findAddressList = ()=>{
        return $http.get("./cart/findAddressList.do");
    };

    //4.提交订单
    this.saveOrder=(order)=>{
        return $http.post("./order/add.do",order);
    }
});